src/theme/BlogPostItem/index.js

```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useState } from 'react';
+import { DiscussionEmbed } from 'disqus-react';
+import BlogPostItem from '@theme-original/BlogPostItem';
+import { useBlogPost } from '@docusaurus/plugin-content-blog/client';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import { useColorMode } from '@docusaurus/theme-common';
+
+export default function BlogPostItemWrapper(props) {
+  const { metadata } = useBlogPost();
+  const { frontMatter, permalink, title } = metadata;
+  const { comments = true } = frontMatter;
+  const { siteConfig, i18n } = useDocusaurusContext();
+  const shortname = siteConfig.customFields.shortname;
+  const { colorMode } = useColorMode();
+  const [key, setKey] = useState(0);
+
+  useEffect(() => {
+    setKey((prevKey) => prevKey + 1);
+  }, [colorMode]);
+
+  return (
+    <>
+      <BlogPostItem {...props} />
+      {comments && (
+        <DiscussionEmbed
+          key={key}
+          shortname={shortname}
+          config={{
+            identifier: permalink,
+            title,
+            language: i18n.currentLocale,
+          }}
+          theme={colorMode}
```

Comment:
This has no effect; there's no `theme` [property for `DiscussionEmbed`](https://github.com/disqus/disqus-react/blob/c07f6c7793f7571bdc08a9575b55f64213450477/src/DiscussionEmbed.jsx#L105-L136). We can remove this.

Reply:
let me recheck